dat/missions/dvaered/dvaered-colonel-transport.lua

```diff
@@ -0,0 +1,160 @@
+   vn.jump("mission description")
+
+   vn.label("mission description")
+   m(fmt.f([["Well, to the point. I need somebody who's not crazy to escort me and my Arsenal to {pnt} in the {sys} system. Would you do that? I can't tell you why, that would be classified information leaked to an outsider. Very dangerous, especially when we can't trust you.]]))
+     {pnt=mem.dest_planet, sys=mem.dest_sys}
+   m(_([["One more thing: another warlord would be only to happy to blow any of the colonel rank such as myself up, so you may need to expect attacks. The ship will probably be called 'Asheron Anomaly', but it may be something else."]]))
+   vn.menu{
+      {_([["I'd be happy to do that!"]]), "sure"},
+      {_([["What is your name?"]]), "what is your name"},
+      {_([["Wait, why is there someone trying to kill you?"]]), "why would you die"},
+   }
+
+   vn.label("what is your name")
+   m(_([[The Dvaered seems slightly taken aback. "Well... that may also be classified information... call me Radver."]]))
+   vn.jump("choice")
+
+   vn.label("sure")
+   m(_([["Surprising but good! I'll be on your ship in a while, so don't leave too soon."]]))
+   vn.func( function ()
+      accepted = true
+   end )
+
+   vn.label("why would you die")
+   m(_([[The Dvaered colonel looks uncomfortable. "That too is classified information. Let's just say that I've got a vendetta (not the ship, the relation) with this other warlord, and we're on blowing-each-other-up terms, and you can't ask me to reveal more information."]]))
+
+   vn.label("choice")
+   m(_([["Well? Will you do this for the sake of all true Dvaered, or else..?"]]))
+   vn.menu{
+      {_([["Yep, I'd be glad to!"]]), "sure"},
+      {_([["Not going to happen, sorry."]]), "never"},
+   }
+
+   vn.label("never")
+   m(_([["Quite an impolite answer," says the Dvaered. "Well, I suppose no choice is offered to me, so I'll stay here, but don't take too long in coming back."]]))
+   vn.done()
+
+   vn.run()
+
+   if not accepted then return end
+
+   misn.accept()
+
+   misn.setReward(750000)
+   misn.setDesc(fmt.f(_("Escort a Dvaered colonel, who is flying an Arsenal, to {pnt} in the {sys} system. You haven't been told why, but there may be a large payment."), {pnt=mem.dest_planet, sys=mem.dest_sys}))
+   misn.osdCreate(_("Dvaered colonel escort"), {
+      fmt.f(_("Escort a Dvaered colonel to {pnt} in the {sys} system.")), {pnt=mem.dest_planet, sys=mem.dest_sys},
+   })
+   misn.markerAdd( mem.dest_planet )
+   hook.land( "land" )
+   local colonel_ship = ship.get("Dvaered Arsenal")
+   escort.init( colonel_ship, {
+      func_pilot_create = function ( p )
+        local fct = ffriendly()
+        p:rename("Radver")
+        local ffriendly = factions()
+        p:setFaction( ffriendly() )
+   end } )
+
+   hook.enter( "ambush" )
+   hook.enter( "factions" )
+end
+
+function ambush ()
+   if not naev.claimTest( system.cur(), true ) then return end
+
+   rm.hook( "ambush" )
```

Comment:
API is `hook.rm` and you would need to pass the id obtained from `hook.enter`. An easier way of doing this is using a global variable. Just do something like:
```lua
if mem.ambushed then return end
mem.ambushed = true
```
Which is easier to do.